api/v1/collector_types.go

```diff
@@ -0,0 +1,64 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CollectorSpec defines the desired state of Collector
+type CollectorSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Type defines the type of collector that this is. Supported values are ecr_event
+	CollectorType string `json:"type"`
+	// QueueName defines the name of the queue to collect events from, in the case of the ecr_event type.
+	QueueName string `json:"queueName"`
```

Comment:
I wonder if this particular attribute should be nested within something that makes it clear that this is specific to AWS.  if/when we end up supporting other ways of collecting in the future, this would get confusing and would require a breaking change to the API down the road.

Reply:
I agree, this was one of the things that stood out to me about the spec as well, but I couldn't think of a great way to structure it. We can't do anything like a conditional field as far as I know.

I suppose just putting another tag per supported type, and nesting under that would be fine.

Reply:
Updated to use a nested field 'ecr'. Should expand more easily for future collector types.